src/lib/gundb/UserStorage.js

```diff
@@ -404,22 +425,31 @@ class UserStorage {
    * need to develop for gundb immutable keys to non first user
    */
   async indexProfileField(field: string, value: string, privacy: FieldPrivacy): Promise<ACK> {
-    if (!UserStorage.indexableFields[field]) return
+    if (!UserStorage.indexableFields[field]) return Promise.resolve({ err: 'Not indexable field', ok: 0 })
     const cleanValue = UserStorage.cleanFieldForIndex(field, value)
-
+    const indexNode = gun.rootAO(`users/by${field}`).get(cleanValue)
     logger.info({ field, value, privacy })
-    if (privacy !== 'public')
-      return gun
-        .get('users')
-        .get('by' + field)
-        .get(cleanValue)
-        .putAck(null)
-
-    const gunResult = await gun
-      .get('users')
-      .get('by' + field)
-      .get(cleanValue)
-      .putAck(this.gunuser)
+
+    const indexValue = await indexNode.then()
+    logger.info({
+      field,
+      value,
+      privacy,
+      indexValue: indexValue,
+      currentUser: this.gunuser.is.pub
+    })
+
+    if (indexValue && indexValue.pub != this.gunuser.is.pub) {
```

Comment:
linter didn't complain about this `!=` instead of `!==`?

We can try to cast one of the operands beforehand